backend/src/cache/server/mutation.go

```diff
@@ -178,14 +191,23 @@ func generateCacheKeyFromTemplate(template string) (string, error) {
 		return "", err
 	}
 
-	// template[archiveLocation] needs to be removed when calculating cache key.
-	// Because archiveLocation.key is different in every single run.
-	_, exists := templateMap[ArchiveLocationKey]
-	if exists {
-		log.Println("ArchiveLocation exists in template.")
-		delete(templateMap, ArchiveLocationKey)
+	// Selectively copying parts of the template that should affect the cache
+	templateSkeleton := map[string]interface{}{
```

Comment:
nit: shall we initiate this as a readonly constant value?

Reply:
Hmm. Are you sure Go support constant maps?

Reply:
I meant, looks like we are not going to modify it, so can we just declare it once as a global variable?